### PR TITLE
Adress issues from review

### PR DESCRIPTION
**Pull request by apeltzer**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/smrnaseq/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/smrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by apeltzer** [comment]
@nf-core-bot fix linting

**Comment by atrigila** [comment]
I can't reproduce the error:

CI: 
image
Gitpod:
image

**Comment by apeltzer** [comment]
CI not working now because of the template v.3.0.0 release. I'm afraid we need to merge thta one in too :-(

**Comment by apeltzer** [comment]
Likely Need to merge in 3.0.1 First and then work on this

**Comment by apeltzer** [comment]
Mergen in `dev` with Template Channels so that wie can fix anything remaining here and then push everything to dev. Hopefully that should do the Trick and we‘re there then .)

**Comment by apeltzer** [comment]
Issue with template will be fixed today with nf-core/tools 3.0.2 —> we should merge that in here directly then.

**Comment by apeltzer** [comment]
Ok next steps would be merging #474 to this PR's branch. Then merging this one to `dev` and ultimately then making the release after updating the changelog date appropriately :)

**Comment by apeltzer** [comment]
Ok should be fine to go

File: .gitignore [hunk 1]
```diff
@@ -6,5 +6,4 @@ results/
 testing/
 testing*
 *.pyc
-execution_trace*
-.nf-test*
```
Comment (maxulysse) [inline .gitignore:10]: don't you want to keep these?

File: CITATIONS.md [hunk 2]
```diff
@@ -38,6 +38,7 @@
 
 - [MultiQC](https://pubmed.ncbi.nlm.nih.gov/27312411/)
 
+> Ewels P, Magnusson M, Lundin S, Käller M. MultiQC: summarize analysis results for multiple tools and samples in a single report. Bioinformatics. 2016 Oct 1;32(19):3047-8. doi: 10.1093/bioinformatics/btw354. Epub 2016 Jun 16. PubMed PMID: 27312411; PubMed Central PMCID: PMC5039924.
 > Ewels P, Magnusson M, Lundin S, Käller M. MultiQC: summarize analysis results for multiple tools and samples in a single report. Bioinformatics. 2016 Oct 1;32(19):3047-8. doi: 10.1093/bioinformatics/btw354. Epub 2016 Jun 16. PubMed PMID: 27312411; PubMed Central PMCID: PMC5039924.
```
Comment (maxulysse) [inline CITATIONS.md:42]:
```suggestion
  > Ewels P, Magnusson M, Lundin S, Käller M. MultiQC: summarize analysis results for multiple tools and samples in a single report. Bioinformatics. 2016 Oct 1;32(19):3047-8. doi: 10.1093/bioinformatics/btw354. Epub 2016 Jun 16. PubMed PMID: 27312411; PubMed Central PMCID: PMC5039924.
```

File: CITATIONS.md [hunk 3]
```diff
@@ -38,6 +38,7 @@
 
 - [MultiQC](https://pubmed.ncbi.nlm.nih.gov/27312411/)
 
+> Ewels P, Magnusson M, Lundin S, Käller M. MultiQC: summarize analysis results for multiple tools and samples in a single report. Bioinformatics. 2016 Oct 1;32(19):3047-8. doi: 10.1093/bioinformatics/btw354. Epub 2016 Jun 16. PubMed PMID: 27312411; PubMed Central PMCID: PMC5039924.
```
Comment (mashehu) [inline CITATIONS.md:41]:
```suggestion
```

File: subworkflows/local/utils_nfcore_smrnaseq_pipeline/main.nf [hunk 4]
```diff
@@ -91,7 +94,7 @@ workflow PIPELINE_INITIALISATION {
         .groupTuple()
         .map { samplesheet ->
             validateInputSamplesheet(samplesheet)
-        }
+            }
```
Comment (maxulysse) [inline subworkflows/local/utils_nfcore_smrnaseq_pipeline/main.nf:97]:
```suggestion
        }
```

File: nextflow.config [hunk 5]
```diff
@@ -90,12 +90,15 @@ params {
     help                         = false
     help_full                    = false
     show_hidden                  = false
+    help_full                    = false
+    show_hidden                  = false
```
Comment (mashehu) [inline nextflow.config:94]:
```suggestion
    help_full                    = false
    show_hidden                  = false
```

**Review by maxulysse — APPROVED** [review]
minor comments

File: subworkflows/local/utils_nfcore_smrnaseq_pipeline/main.nf [hunk 6]
```diff
@@ -152,15 +157,49 @@ workflow PIPELINE_COMPLETION {
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 //
 // Check and validate pipeline parameters
 //
 def validateInputParameters() {
     genomeExistsError()
+
+    if (!params.mirgenedb && !params.mirtrace_species) {
+            error("Reference species for miRTrace is not defined via the --mirtrace_species parameter.")
```
Comment (mashehu) [inline subworkflows/local/utils_nfcore_smrnaseq_pipeline/main.nf:173]:
```suggestion
            error("Reference species for miRTrace is not defined via the `--mirtrace_species` parameter.")
```

File: subworkflows/local/utils_nfcore_smrnaseq_pipeline/main.nf [hunk 7]
```diff
@@ -152,15 +157,49 @@ workflow PIPELINE_COMPLETION {
...
+    }
+
+    if (!params.mirgenedb) {
+        // Validate mature miRNA fasta file
+        if (!params.mature) {
+            error("Mature miRNA fasta file not found. Please specify using the '--mature' parameter.")
```
Comment (mashehu) [inline subworkflows/local/utils_nfcore_smrnaseq_pipeline/main.nf:179]:
```suggestion
            error("Mature miRNA fasta file not found. Please specify using the `--mature` parameter.")
```
please do the same to the rest.

**Review by mashehu — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (apeltzer) [reply .gitignore:10]: I do ! ;-)

**Pull request merged**